src/AutoMapper/Mapper.cs

```diff
@@ -186,11 +186,11 @@ TDestination IMapper.Map<TDestination>(object source)
         {
             var types = new TypePair(source.GetType(), typeof(TDestination));
 
-            dynamic func = _configurationProvider.GetMapperFunc(types);
+            var func = _configurationProvider.GetMapperFunc(types);
 
             var context = new ResolutionContext(source, null, source.GetType(), typeof(TDestination), null, _defaultMappingOptions, this);
 
-            return func((dynamic)source, null, context);
+            return (TDestination) func.DynamicInvoke(source, null, context);

```

Comment:
Yep, I'm hoping to fix this in a later PR. Ideally it's only external Mapper calls that use this, not internal ones....eventually.